src/lib/viewers/media/DashViewer.js

```diff
@@ -402,14 +447,25 @@ class DashViewer extends VideoBaseViewer {
 
         // Iterate over all available video representations and find the one that
         // seems the biggest so that the video player is set to the max size
-        const hdRepresentation = tracks.reduce((a, b) => (a.width > b.width ? a : b));
-        const sdRepresentation = tracks.reduce((a, b) => (a.width < b.width ? a : b));
+        const hdRep = tracks.reduce((a, b) => (a.width > b.width ? a : b));
+        const sdRep = tracks.reduce((a, b) => (a.width < b.width ? a : b));
+        if (sdRep.videoId === null) {
+            // There is only audio, no video
+            this.videoWidth = DEFAULT_VIDEO_WIDTH;
+            this.videoHeight = DEFAULT_VIDEO_HEIGHT;
+        } else if (hdRep.videoId === sdRep.videoId) {
+            // Only one video quality
+            this.videoWidth = sdRep.width;
+            this.videoHeight = sdRep.height;
+            this.sdVideoId = sdRep.videoId;
+        } else {
+            this.videoWidth = hdRep.width;
+            this.videoHeight = hdRep.height;
+            this.hdVideoId = hdRep.videoId;
+            this.sdVideoId = sdRep.videoId;
```

Comment:
Could be simplified:

```
if (sdRep.videoId === null) {
    // There is only audio, no video
    this.videoWidth = DEFAULT_VIDEO_WIDTH;
    this.videoHeight = DEFAULT_VIDEO_HEIGHT;
} else {
    this.videoWidth = hdRep.width;
    this.videoHeight = hdRep.height;
    this.hdVideoId = hdRep.videoId !== sdRep.videoId ? hdRep.videoId : undefined;
    this.sdVideoId = sdRep.videoId;
}
```

Reply:
Thanks. Also shaka 2.2.0 has an isAudioOnly() method, which I can call now instead of checking sdRep.videoId === null